File: TFGWordle/src/main/java/app/TFGWordle/dto/WordleState.java
package app.TFGWordle.dto;

import java.util.List;

public class WordleState {
    private int numberOfWordle;
    private List<Integer> wordleOrder;
    private List<Game> games;

    public int getNumberOfWordle() {
        return numberOfWordle;
    }

    public void setNumberOfWordle(int numberOfWordle) {
        this.numberOfWordle = numberOfWordle;
    }

    public List<Integer> getWordleOrder() {
        return wordleOrder;
    }

    public void setWordleOrder(List<Integer> wordleOrder) {
        this.wordleOrder = wordleOrder;
    }

    public List<Game> getGames() {
        return games;
    }

    public void setGames(List<Game> games) {
        this.games = games;
    }

    public static class Game {
        private boolean finished;
        private boolean won;
        private int tryCount;
        private String startTime;
        private String finishTime;
        private int timeNeeded;
        private State state;
        private String lastWordle;
        private String timeGuess;

        public boolean isFinished() {
            return finished;
        }

        public void setFinished(boolean finished) {
            this.finished = finished;
        }

        public boolean isWon() {
            return won;
        }

        public void setWon(boolean won) {
            this.won = won;
        }

        public int getTryCount() {
            return tryCount;
        }

        public void setTryCount(int tryCount) {
            this.tryCount = tryCount;
        }

        public String getStartTime() {
            return startTime;
        }

        public void setStartTime(String startTime) {
            this.startTime = startTime;
        }

        public String getFinishTime() {
            return finishTime;
        }

        public void setFinishTime(String finishTime) {
            this.finishTime = finishTime;
        }

        public int getTimeNeeded() {
            return timeNeeded;
        }

        public void setTimeNeeded(int timeNeeded) {
            this.timeNeeded = timeNeeded;
        }

        public State getState() {
            return state;
        }

        public void setState(State state) {
            this.state = state;
        }

        public String getLastWordle() {
            return lastWordle;
        }

        public void setLastWordle(String lastWordle) {
            this.lastWordle = lastWordle;
        }

        public String getTimeGuess() {
            return timeGuess;
        }

        public void setTimeGuess(String timeGuess) {
            this.timeGuess = timeGuess;
        }

        public static class State {
            private int good;
            private int halfGood;
            private int wrong;

            public int getGood() {
                return good;
            }

            public void setGood(int good) {
                this.good = good;
            }

            public int getHalfGood() {
                return halfGood;
            }

            public void setHalfGood(int halfGood) {
                this.halfGood = halfGood;
            }

            public int getWrong() {
                return wrong;
            }

            public void setWrong(int wrong) {
                this.wrong = wrong;
            }
        }
    }
}

File: frontend/src/app/models/wordle-state.ts
export class WordleState {
  numberOfWordle: number;
  games: Game[];
  wordleOrder: number[];

  constructor(numberOfWordle: number = 0, games: Game[] = [], isRandom: boolean) {
    this.numberOfWordle = numberOfWordle;
    this.games = games;
    this.wordleOrder = isRandom
      ? this.shuffle(Array.from({ length: numberOfWordle }, (_, i) => i))
      : Array.from( { length: numberOfWordle }, (_, i) => i);
  }

  private shuffle(array: number[]): number[] {
    for (let i = array.length - 1; i > 0; i--) {
      const j = Math.floor(Math.random() * (i + 1));
      [array[i], array[j]] = [array[j], array[i]];
    }
    return array;
  }
}

export class Game {
  finished: boolean;
  won: boolean;
  tryCount: number;
  state: State;
  startTime: string;
  finishTime: string;
  timeNeeded: number;
  lastWordle: string;
  timeGuess: string;

  constructor(finished: boolean = false, won: boolean = false, tryCount: number = 0, state: State = new State(), startTime: string = '', finishTime: string = '', timeNeeded: number = 0, lastWordle: string = "", timeGuess: string = "") {
    this.finished = finished;
    this.won = won;
    this.tryCount = tryCount;
    this.state = state;
    this.startTime = startTime;
    this.finishTime = finishTime;
    this.timeNeeded = timeNeeded;
    this.lastWordle = lastWordle;
    this.timeGuess = timeGuess;
  }
}

export class State {
  good: number;
  halfGood: number;
  wrong: number;

  constructor(good: number = 0, halfGood: number = 0, wrong: number = 0) {
    this.good = good;
    this.halfGood = halfGood;
    this.wrong = wrong;
  }
}
